Named terminals, indicate active terminal, close active terminal

BenWildeman opened the pull request:
fixes #10 #6 

Also adds the ability to close the terminal by clicking the active status bar button again

![gif](http://i.imgur.com/sLLdIed.gif)

Commit:
named terminals

BenWildeman committed:
Show active terminal and add the ability to close active terminal

Tyriar commented on src/extension.ts:
```diff
@@ -18,21 +18,29 @@ export function activate(context: vscode.ExtensionContext) {
             placeHolder: 'Enter the name of the new terminal'
         }).then(name => {
+            for (let j = 0; j <_terminals.length; j++) {
+                _terminals[j].hide();
```
I think this may lead to a flicker from the panel hiding and then showing immediately. Maybe this should call `StatusBarTerminal.markHidden()` which is also called by `StatusBarTerminal .hide()`?

Tyriar commented on src/statusBarTerminal.ts:
```diff
@@ -2,24 +2,41 @@ import * as vscode from 'vscode';
 
     public show() {
+        this._showing = true;
+        this._item.color = "yellow";
```
Maybe this color should be a setting? Some users may not like the very bright yellow.

Tyriar commented on src/statusBarTerminal.ts:
```diff
@@ -2,24 +2,41 @@ import * as vscode from 'vscode';
+    }
+
+    public setTerminalIndex(i: number, name?: string) {
+        this._item.text = `$(terminal) ${name ? name: (i + 1)}`;
```
Can you put a space before the `:`?

Tyriar commented on src/extension.ts:
```diff
@@ -18,21 +18,29 @@ export function activate(context: vscode.ExtensionContext) {
     for (let i = 1; i <= MAX_TERMINALS; i++) {
         context.subscriptions.push(vscode.commands.registerCommand(`terminalTabs.showTerminal${i}`, (a) => {
-            _terminals[i - 1].show();
+            for (let j = 0; j <_terminals.length; j++) {
```
_terminals.forEach would probably be better for readability/consistency as performance is no issue in this extension.

Tyriar reviewed (changes requested):
Looks good! I just had a few comments.

BenWildeman committed:
Update extension as per suggestions on pull request

BenWildeman commented:
sheesh didn't realise that it was 3 months ago since I did this pull request. time flies. I have updated the things you have suggested. with the colour setting, should this be added to the readme? would you like me to update it with this pull request or are you happy to add this yourself?

I have also updated the typescript/vscode versions as per https://code.visualstudio.com/updates/v1_6#_extension-authoring

BenWildeman committed:
change for loop to forEach

BenWildeman committed:
added back coment

jing2si commented:
This is a very useful feature. Please merge.

akaibukai commented:
@BenWildeman Hello!
I wanted to submit an issue about why not having the name of the terminal appearing instead of an index, especially that there is an option to have named terminal..
Then I noticed your version..
At first I didn't noticed it, on the extension search results (maybe because of the lower download counts?).
But your extension rocks!
Thank you!

Tyriar commented:
FYI I'm not planning on working further on this extension as it's just a temporary workaround until we actually get tabs with https://github.com/microsoft/vscode/issues/10546

akaibukai commented:
Hi,
Since it's currently doing the job, I'm happy if it's no more maintained.
I even discovered that I can preset terminals with dedicated path and optional command in the settings. And using a workspace setting file I can even set this per project which is really great!

I take a look at the VS Code issue... It's almost 4 years old! That's a huge timespan for such an important feature imho..

But I guess it's not that simple particularly to have bug free features regarding the huge ecosystem it exists now...

And thank you for taking the time to inform..
